Rock_Paper_Scissor/Rock_Paper_Scissor.java

```diff
@@ -0,0 +1,123 @@
+import java.util.Scanner;
+public class Main
+{
+  public static int large = 500;
+  public static double dp[][][] = new double[large][large][large];
+
+  public static double probrock (int r, int p, int s)
+  {
+    if (s == 0 || r == 0)
+      return 0.0;
+
+    if (p == 0)
+      return 1.0;
+
+    if (dp[r][p][s] != -1.0)
+      return dp[r][p][s];
+
+    //total no. of fight = rC1.pC1 + pC1.sC1 + sC1.rC1
+    double total = r * p + p * s + s * r;
+    double res = 0.0;
+
+      res += probrock (r - 1, p, s) * (((double) (r * p)) / total);	//prob. of fight btw r&p = r*p/total
+      res += probrock (r, p - 1, s) * (((double) (s * p)) / total);	//prob. of fight btw s&p = s*p/total
+      res += probrock (r, p, s - 1) * (((double) (r * s)) / total);	//prob. of fight btw r&s= r*s/total 
+
+      return dp[r][p][s] = res;
+  }
+
+  public static double probpaper (int r, int p, int s)
+  {
+    if (p == 0 || r == 0)
+      return 0.0;
+
+    if (s == 0)
+      return 1.0;
+
+    if (dp[r][p][s] != -1.0)
+      return dp[r][p][s];
+
+    //total no. of fight = rC1.pC1 + pC1.sC1 + sC1.rC1
+    double total = r * p + p * s + s * r;
+    double res = 0.0;
+
+    res += probpaper (r - 1, p, s) * (((double) (r * p)) / total);
+    res += probpaper (r, p - 1, s) * (((double) (s * p)) / total);
+    res += probpaper (r, p, s - 1) * (((double) (r * s)) / total);
+
+    return dp[r][p][s] = res;
+  }
+
+  public static double probscissor (int r, int p, int s)
+  {
+    if (p == 0 || s == 0)
+      return 0.0;
+
+    if (r == 0)
+      return 1.0;
+
+    if (dp[r][p][s] != -1.0)
+      return dp[r][p][s];
+
+    //total no. of fight = rC1.pC1 + pC1.sC1 + sC1.rC1
+    double total = r * p + p * s + s * r;
+    double res = 0.0;
+
+    res += probscissor (r - 1, p, s) * (((double) (r * p)) / total);
+    res += probscissor (r, p - 1, s) * (((double) (s * p)) / total);
+    res += probscissor (r, p, s - 1) * (((double) (r * s)) / total);
+
+    return dp[r][p][s] = res;
+  }
+
+  public static void main (String[]args)
+  {
+    int rock, scissor, paper;
```

Comment:
Please correct the indentation(4 space indentation)

Reply:
@somya-kapoor I've made the changes, please check the file